### PR TITLE
adding tractions methods to cython and unit tests

### PR DESCRIPTION
**Pull request by timryanb**

### CODE REVIEW

File: src/elements/TACSElement.h [hunk 1]
```diff
@@ -183,6 +183,15 @@ class TACSElement : public TACSObject {
     return NULL;
   }
 
+  /**
+    Create element traction class
+
+    @return The TACSElement traction class associated with this element. Possibly NULL.
+  */
+  virtual TACSElement* createElementTraction(TacsScalar t[], int faceIndex){
```
Comment (gjkennedy) [inline src/elements/TACSElement.h:191]: I think the faceIndex argument should come first to be consistent with the way we've typically written these function prototypes.

File: src/elements/TACSElement2D.cpp [hunk 2]
```diff
@@ -51,6 +52,11 @@ TACSElementModel* TACSElement2D::getElementModel(){
   return model;
 }
 
+TACSElement* TACSElement2D::createElementTraction(TacsScalar t[], int faceIndex){
```
Comment (gjkennedy) [inline src/elements/TACSElement2D.cpp:55]: Please put a space before the first variable type and after the last argument

Comment (gjkennedy) [reply src/elements/TACSElement2D.cpp:55]: Same throughout

File: src/TACSAssembler.cpp [hunk 3]
```diff
@@ -4898,21 +4898,27 @@ void TACSAssembler::addAdjointResProducts( TacsScalar scale,
     }
 
     // Add the contribution from the auxiliary elements
-    while (aux_count < naux && aux[aux_count].num == i){
-      // Get the design variables for this element
-      numDVs = aux[aux_count].elem->getDesignVarNums(i, maxDVs, dvNums);
+    if (aux_count < naux){
```
Comment (gjkennedy) [inline src/TACSAssembler.cpp:4901]: Thanks for catching this!

File: src/elements/TACSElement.h (hunk 1 above)
Comment (timryanb) [reply src/elements/TACSElement.h]: Should be fixed now

**Review by gjkennedy — APPROVED** [review]

**Pull request merged**